serviceregistration/kubernetes/client/client.go

```diff
@@ -247,6 +248,40 @@ func (e *ErrNotFound) Error() string {
 	return e.debuggingInfo
 }
 
+// Sanitize is for "data" being sent to the Kubernetes API.
+// Data must consist of alphanumeric characters, '-', '_' or '.'.
+// Any other characters found in the original value will be stripped,
+// and the surrounding characters will be concatenated.
+func Sanitize(val string) string {
+	allowable := ""
+	for _, ru := range val {
```

Comment:
This seems like a good place for `strings.Map`

Reply:
e.g. something like:
```
	return strings.Map(func(r rune) rune {
		if isAllowable(r) {
			return r
		}
		return '-'
	}
```

Reply:
I gave that idea a try, but since the Map requires a func that returns _one_ rune, the code gets awkward when converting the string `"-"` into a rune. It's okay to do this `[]rune("-")`, but not without it being a rune array. If I did it that way I'd need to then draw out just one rune to return, and to me that seems strange. Do you know a better way?

Reply:
Thanks for the tip about " vs '. https://github.com/hashicorp/vault/pull/8411/commits/aba6a204966ca78dd44fc0240bb71e9fb258074b